app/assets/javascripts/application.js

```diff
@@ -17,11 +17,84 @@
 //= require turbolinks
 //= require_tree .
 
+(function ($) {
+    $.fn.longClick = function (callback, timeout) {
+        var timer;
+        timeout = timeout || 500;
+        $(this).mousedown(function () {
+            timer = setTimeout(function () {
+                callback();
+            }, timeout);
+            return false;
+        });
+        $(document).mouseup(function () {
+            clearTimeout(timer);
+            return false;
+        });
+    };
 
-function addRows(count) {
-    for (i = 0; i < count; i++) {
-        $(".card .content").append('<div class="action" id="action_' + i + '">Slot ' + (i + 1) + '</div>');
-    }
+})(jQuery);
+
+
+function queryApi(date) {
+    $.ajax({
+        dataType: "json",
+        url: '/api_index.json?date=' + date,
+        success: function (response) {
+            loadCurrentBookings(response);
+            addEvents();
+        }
+    });
+}
+
+function updateApi(date) {
+    $.ajax({
+        dataType: "json",
+        url: '/api_index.json?date=' + date,
+        success: function (response) {
+            updateCurrentBookings(response);
+            addEvents();
+        }
+    });
+}
+
+
+function loadCurrentBookings(response) {
+    var res = response;
+    res.items.forEach(function (item) {
+        var card = ['#card', item.id].join('-');
+        $(card + " .content").append('<div class="with-scroll"></div>')
+        item.slots.forEach(function (slot) {
+            $(card + " .content .with-scroll").append('<div class="action" id="action_' + slot.info.id + '" style="background-color: ' + getBackgroundColor(slot) + '">' + getInfo(slot) + '</div>');
+        });
+    });
+}
+
+function updateCurrentBookings(response) {
+    var res = response;
+    res.items.forEach(function (item) {
+        var card = ['#card', item.id].join('-');
+        $(card + " .content .with-scroll").empty();
+        item.slots.forEach(function (slot) {
+            $(card + " .content .with-scroll").append('<div class="action" id="action_' + slot.info.id + '" style="background-color: ' + getBackgroundColor(slot) + '">' + getInfo(slot) + '</div>');
+        });
+    });
+
+}
+
```

Comment:
The `loadCurrentBookings()` and `updateCurrentBookings()` functions could be refactored into one single method.